IPython/terminal/shortcuts.py

```diff
@@ -123,6 +130,20 @@ def _following_text():
         _following_text_cache[pattern] = condition
         return condition
 
+    @Condition
+    def not_inside_unclosed_string():
+        app = get_app()
+        preceding_text = app.current_buffer.document.current_line_before_cursor
+        if parso_loaded:
```

Comment:
this is _one_ reason we'd need to use the fallback, but if some _other_ substantial API change is made, those uses of the API, members, etc. might change, so this may need a `try/except` block around all the uses of the parso api, with a duplicate `return True`.

I wouldn't be so picky about this, but the last time we had to deal with a _minor_ jedi/parso API change, it took _months_ to resolve properly.